source/module_hamilt_pw/hamilt_pwdft/forces_scc.cpp

```diff
@@ -0,0 +1,233 @@
+#include "forces.h"
+#include "module_hamilt_pw/hamilt_pwdft/global.h"
+#include "module_io/output_log.h"
+#include "stress_func.h"
+// new
+#include "module_base/complexmatrix.h"
+#include "module_base/libm/libm.h"
+#include "module_base/math_integral.h"
+#include "module_base/mathzone.h"
+#include "module_base/timer.h"
+#include "module_base/tool_threading.h"
+#include "module_elecstate/potentials/efield.h"
+#include "module_elecstate/potentials/gatefield.h"
+#include "module_hamilt_general/module_ewald/H_Ewald_pw.h"
+#include "module_hamilt_general/module_surchem/surchem.h"
+#include "module_hamilt_general/module_vdw/vdw.h"
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+#ifdef USE_PAW
+#include "module_cell/module_paw/paw_cell.h"
+#endif
+
+
+template <typename FPTYPE, typename Device>
+void Forces<FPTYPE, Device>::cal_force_scc(ModuleBase::matrix& forcescc,
+                                           ModulePW::PW_Basis* rho_basis,
+                                           const ModuleBase::matrix& vnew,
+                                           const bool vnew_exist,
+                                           const UnitCell& ucell_in) {
+    ModuleBase::TITLE("Forces", "cal_force_scc");
+    ModuleBase::timer::tick("Forces", "cal_force_scc");
+
+    // for orbital free case
+    if (!vnew_exist) {
+        ModuleBase::timer::tick("Forces", "cal_force_scc");
+        return;
+    }
+
+    std::complex<double>* psic = new std::complex<double>[rho_basis->nmaxgr];
```

Comment:
could you use std::vector to replace new and delete?

Reply:
sure